src/lib.rs

```diff
@@ -391,22 +392,28 @@ impl OptUdeps {
 }
 
 struct ExecData {
-	cargo_exe :Option<String>,
+	cargo_exe :OsString,
 	supports_color :bool,
 	relevant_cmd_infos :Vec<CmdInfo>,
 }
 
 impl ExecData {
-	fn new(supports_color :bool) -> Self {
-		let cargo_exe = which::which("cargo")
-			.map_err(|e| println!("warning: couldn't load cargo executable file: {:?}", e))
-			.ok()
-			.and_then(|p| p.to_str().map(str::to_owned));
-		Self {
+	fn new(config :&Config) -> CargoResult<Self> {
+		// `$CARGO` should be present when `cargo-udeps` is executed as `cargo udeps ..` or `cargo run -- udeps ..`.
+		let cargo_exe = env::var_os(cargo::CARGO_ENV)
+			.map(Ok::<_, failure::Error>)
+			.unwrap_or_else(|| {
+				// Unless otherwise specified, `$CARGO` is set to `config.cargo_exe()` for compilation commands which points at `cargo-udeps`.
+				let cargo_exe = config.cargo_exe()?;
+				config.shell().warn(format!("$CARGO is set to {}", cargo_exe.display()))?;
```

Comment:
Can you change it to `Couldn't find $CARGO environment variable. Setting it to {}`

Reply:
Ok.